src/Actions/Bulk/LockThreads.php

```diff
@@ -28,8 +28,10 @@ protected function transact()
             $query = $query->whereNull(Thread::DELETED_AT);
         }
 
+        $threads = $query->get();
+
         // Return early if there are no eligible threads in the selection
-        if ($query->count() == 0) return null;
+        if ($threads->count() == 0) return null;
 
         $query->update(['locked' => true]);
```

Comment:
Would this be better... ?
```
         // Return early if there are no eligible threads in the selection
         if ($query->count() == 0) return null;

         $query->update(['locked' => true]);

         return $threads = $query->get();
```
No extra operation, just performs the get after the update - this way you return the current state of the threads, not the state pre-update.

Could be done as a separate PR.

Reply:
That would produce another query - `$threads->count()` counts the items in the collection and doesn't hit the DB.

I think a separate PR to address the state problem by checking the number of affected rows and fetching them afterwards instead of fetching them before the update would be good.